sys/vistaff

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+from argparse import ArgumentParser
+from fcntl import flock
+from fcntl import LOCK_EX
+from fcntl import LOCK_NB
+from fcntl import LOCK_UN
+
+import yaml
+from ocflib.misc.shell import edit_file
+from staff_hours import get_staff_hours
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument('path')
+    args = parser.parse_args()
+
+    content = None
+    with open(args.path, 'r') as target:
+        content = target.read()
+
+    with open(args.path, 'r+') as target:
+        try:
+            flock(target.fileno(), LOCK_EX | LOCK_NB)
+        except IOError:
+            print('Somebody else is modifying this file!')
```

Comment:
Would it be possible to print who is modifying the file, and perhaps how long they've been editing it for? (ie if someone has been editing for 10 hours then there's probably a mistake)

Reply:
I guess we can revisit this if it actually becomes a problem (which I suspect it will be) but for now I'm fine if this doesn't have it.

Reply:
Not pretty, but it'll work.
```
import subprocess
def get_access_time():
    pid = subprocess.check_output(['pgrep', '-o', 'easywp'])
    pid = pid.decode().splitlines()[0]
    time = subprocess.check_output(['ps', '-o', 'etime', '-o', 'ruser=RealUser', pid])
    pdata = time.decode().splitlines()[1].strip()
    time, user = pdata.split()
    return user, time

user, time = get_access_time()
print('{} has been modifying the file for {}'.format(user, time))
```